resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ResourceTest.java

```diff
@@ -1127,6 +1127,7 @@ protected void setUp() throws Exception {
 
 	@Override
 	protected void tearDown() throws Exception {
+		assertAndRestoreStateOfJobManager();
```

Comment:
Sorry that I did not see and comment that one before: asserting the state before doing the rest of the cleanup will leave the system in an inconsistent state after this test in case the job manager was not resumed properly. The assertion should be moved to after every other kind of cleanup.

Reply:
Good catch, I hadn't thought about that. I moved the assertion all the way to the end to let the test clean up the rest properly before failing (in 656c74ac8f5aa0509d59604d4d70b03ce492601a)